glif agent pay to-current

I only borrowed 1 FIL... so the panic is one issue but the payment of 2302130898021 FIL is another.
```
❯ glif agent pay to-current
2023/05/16 12:11:48 Paying 2302130898021 FIL to the infinity-pool
-Transaction: 0x4d85ec0cc11d60e24c7ff95571a1493959df5ae42541069679dbccd7c65867fd
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/glif-confidential/cli/cmd.glob..func14(0x101599fe0, {0x1015db3e0, 0x0, 0x0?})
	/Users/lanzafame/src/github.com/glif-confidential/cli/cmd/agent_pay_current.go:56 +0x49f
github.com/spf13/cobra.(*Command).execute(0x101599fe0, {0x1015db3e0, 0x0, 0x0})
	/Users/lanzafame/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x101599740)
	/Users/lanzafame/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/lanzafame/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/glif-confidential/cli/cmd.Execute()
	/Users/lanzafame/src/github.com/glif-confidential/cli/cmd/root.go:46 +0x25
main.main()
	/Users/lanzafame/src/github.com/glif-confidential/cli/main.go:21 +0x17
```

--- a/cmd/agent_id.go
+++ b/cmd/agent_id.go
@@ -22,13 +22,13 @@ var idCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
+		log.Printf("Fetching agent ID for %s", util.TruncateAddr(agentAddr.String()))
 		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 		s.Start()
 
-		log.Printf("Fetching agent ID for %s", util.TruncateAddr(agentAddr.String()))
-
 		id, err := PoolsSDK.Query().AgentID(cmd.Context(), agentAddr)
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
 

--- a/cmd/agent_info.go
+++ b/cmd/agent_info.go
@@ -48,11 +48,13 @@ var agentInfoCmd = &cobra.Command{
 
 		version, err := query.AgentVersion(cmd.Context(), agentAddr)
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
 
 		assets, err := query.AgentLiquidAssets(cmd.Context(), agentAddr)
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
 
@@ -60,6 +62,7 @@ var agentInfoCmd = &cobra.Command{
 
 		lvl, cap, err := query.InfPoolGetAgentLvl(cmd.Context(), agentID)
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
 
@@ -79,18 +82,23 @@ var agentInfoCmd = &cobra.Command{
 
 		account, err := query.InfPoolGetAccount(cmd.Context(), agentAddr)
 		if err != nil {
+			s.Stop()
 			log.Fatalf("Failed to get iFIL balance %s", err)
 		}
 
 		defaultEpoch, err := query.DefaultEpoch(cmd.Context())
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
 
 		amountOwed, gcred, err := query.AgentOwes(cmd.Context(), agentAddr)
 		if err != nil {
+			s.Stop()
 			log.Fatal(err)
 		}
+
+		s.Stop()
 
 		amountOwedFIL, _ := util.ToFIL(amountOwed).Float64()
 

--- a/cmd/agent_pay_current.go
+++ b/cmd/agent_pay_current.go
@@ -37,7 +37,7 @@ var payToCurrentCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		log.Printf("Paying %s FIL to the %s", amountOwed, poolName)
+		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed), poolName)
 
 		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 		s.Start()

--- a/cmd/agent_pay_current.go
+++ b/cmd/agent_pay_current.go
@@ -6,9 +6,11 @@ package cmd
 import (
 	"fmt"
 	"log"
+	"math/big"
 	"time"
 
 	"github.com/briandowns/spinner"
+	"github.com/glifio/go-pools/util"
 	"github.com/spf13/cobra"
 )
 
@@ -53,7 +55,10 @@ var payToCurrentCmd = &cobra.Command{
 
 		s.Stop()
 
-		fmt.Printf("Successfully paid %s FIL", args[0])
+		paidAmount := new(big.Int)
+		paidAmount.SetString(args[0], 10)
+
+		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount))
 	},
 }
 

--- a/cmd/agent_pay_current.go
+++ b/cmd/agent_pay_current.go
@@ -6,7 +6,6 @@ package cmd
 import (
 	"fmt"
 	"log"
-	"math/big"
 	"time"
 
 	"github.com/briandowns/spinner"
@@ -37,7 +36,7 @@ var payToCurrentCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed), poolName)
+		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed).String(), poolName)
 
 		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 		s.Start()
@@ -55,10 +54,17 @@ var payToCurrentCmd = &cobra.Command{
 
 		s.Stop()
 
-		paidAmount := new(big.Int)
-		paidAmount.SetString(args[0], 10)
+		if len(args) == 0 {
+			fmt.Printf("Successfully paid %s FIL", util.ToFIL(amountOwed).String())
+			return
+		}
 
-		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount))
+		paidAmount, err := parseFILAmount(args[0])
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount).String())
 	},
 }
 

--- a/cmd/agent_pay_principal.go
+++ b/cmd/agent_pay_principal.go
@@ -44,7 +44,7 @@ var payPrincipalCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		log.Printf("Paying %s FIL to the %s", amountOwed, poolName)
+		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed), poolName)
 
 		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 		s.Start()

--- a/cmd/agent_pay_principal.go
+++ b/cmd/agent_pay_principal.go
@@ -44,7 +44,7 @@ var payPrincipalCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed), poolName)
+		log.Printf("Paying %s FIL to the %s", util.ToFIL(amountOwed).String(), poolName)
 
 		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 		s.Start()
@@ -62,10 +62,12 @@ var payPrincipalCmd = &cobra.Command{
 
 		s.Stop()
 
-		paidAmount := new(big.Int)
-		paidAmount.SetString(args[0], 10)
+		paidAmount, err := parseFILAmount(args[0])
+		if err != nil {
+			log.Fatal(err)
+		}
 
-		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount))
+		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount).String())
 	},
 }
 

--- a/cmd/agent_pay_principal.go
+++ b/cmd/agent_pay_principal.go
@@ -10,6 +10,7 @@ import (
 	"time"
 
 	"github.com/briandowns/spinner"
+	"github.com/glifio/go-pools/util"
 	"github.com/spf13/cobra"
 )
 
@@ -61,7 +62,10 @@ var payPrincipalCmd = &cobra.Command{
 
 		s.Stop()
 
-		fmt.Printf("Successfully paid %s FIL", args[0])
+		paidAmount := new(big.Int)
+		paidAmount.SetString(args[0], 10)
+
+		fmt.Printf("Successfully paid %s FIL", util.ToFIL(paidAmount))
 	},
 }
 

